app/controllers/api/v1/authentication_controller.rb

```diff
@@ -167,6 +167,23 @@ def register_device
         render nothing: true, status: 204
       end
 
+      api :POST, '/v1/auth/register_user_and_send_pin', "Search User by mobile number and create if user not found."
+      def register_user_and_send_pin
```

Comment:
I think we can use the ```signup``` endpoint here. You have ```User.find_or_create_user``` but this is doing the same as ```User.creation_with_auth``` above. The validation for mobile is handled inside the User model.

Reply:
Hi @steveyken , 

Yup.. We also thought about it and its good suggestion. We cancelled using it due to params difference as we were only sending mobile number in params instead of user_params but I think we can send user_params and use existing sign_up endpoint. Will collaborate with @abulsayyad123 for same.

Thanks.